python/pyarrow/tests/test_scalars.py

```diff
@@ -215,3 +215,21 @@ def test_array_to_set(self):
         set_from_array = set(arr)
         assert isinstance(set_from_array, set)
         assert set_from_array == {1, 2}
+
+    def test_struct_array_subscripting(self):
```

Comment:
Nit: call this `test_struct_value_subscripting`? You're not subscripting the array here.